### PR TITLE
Feat/15828803 action fields

### PR DESCRIPTION
**Pull request by CzarekDryl**
https://tw.auroracreation.com/app/tasks/15828803

**New stuff** ✨

* Added fields to created action - draft mode and completed action

## TODO

- Custom action title

## Out of scope

- Right side of action sidebar

Resolves [https://github.com/JoinColony/colonyCDapp/issues/2044](https://github.com/JoinColony/colonyCDapp/issues/2044)

### CODE REVIEW

**Comment by jakubcolony** [comment]
@arrenv @AdrianDudko are we missing a corresponding Github issue? I'm not sure of the scope of this ticket and can't access the aurora creation link.

**Comment by CzarekDryl** [comment]
@jakubcolony We do have issue but it applies to the whole task, not just part of it, because we make the subtasks ourselves.[https://github.com/JoinColony/colonyCDapp/issues/2005](https://github.com/JoinColony/colonyCDapp/issues/2005)

**Comment by arrenv** [comment]
> @jakubcolony We do have issue but it applies to the whole task, not just part of it, because we make the subtasks ourselves.#2005

@CzarekDryl It would be good to copy the relevant acceptance criteria into it's own issue that can be referenced.

**Comment by CzarekDryl** [comment]
@arrenv @jakubcolony New issue added

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Thanks for adding the related issue @CzarekDryl.

It looks like this PR needs rebasing but I ran some high-level functional testing. A few things I noticed:

- Action type is `Generic` instead of `Payment builder`:
Screenshot 2024-03-14 at 22 30 03

- It is possible to crash the app when you create a payment with some specific amounts (e.g. `5342534`) which is most likely due to JS float precision, we need to be able to handle it though:


https://github.com/JoinColony/colonyCDapp/assets/112586815/2e369ba7-b6da-4084-a0a7-adf03fd67d12

**Comment by CzarekDryl** [comment]
@jakubcolony I think that I've fixed that issue with decimals, please check it again.

File: src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts
Comment (jakubcolony) [inline src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts:257]: Can also use the `getSelectedToken` here

File: src/components/v5/common/ActionSidebar/ActionSidebar.tsx
Comment (jakubcolony) [inline src/components/v5/common/ActionSidebar/ActionSidebar.tsx:45]: This hook seems to be called in `useGetActionData` as well, so maybe whatever data you need here could be exported from that hook instead?

File: src/components/v5/common/ActionSidebar/hooks/useGetExpenditureData.ts
Comment (jakubcolony) [inline src/components/v5/common/ActionSidebar/hooks/useGetExpenditureData.ts:10]: The query shouldn't just run with an empty id, please make sure to specify `skip` option to avoid unnecessary requests

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Thanks for your work @CzarekDryl, left a couple of comments in code.

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @CzarekDryl thanks for your work on this PR.

Every time I try a new transaction to test I just get caught in this loading state. I've tried opening it again using the transaction URL and get the same loading state. It was using permissions.

image

If I try using reputation, I can't get past the pending state. 

image

Maybe I'm not doing something correctly here?

File: amplify/backend/api/colonycdapp/schema.graphql
Comment (jakubcolony) [inline amplify/backend/api/colonycdapp/schema.graphql:756]: Looks like you've pulled my changes from #1969 - that means `TEMP_ADVANCED_PAYMENT` isn't needed anymore

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Looking very solid @CzarekDryl, thanks for your changes.

I left some comments in code, and also noticed the following:

- The table breaks quite a bit shortly before going mobile, also notice how the description breaks into multiple lines but `leela` stays aligned to the right:
image

- Overlapping issue on larger screen width with longer amounts:
image

- Neither the recipients nor the tokens are clickable which feels inconsistent with every other instance of user/token link that opens a popover. But I'm not sure what the requirements say about this so one for @melyndav to confirm maybe? 🙂 

- I also thought I saw the payments table load without any rows but I wasn't able to replicate that - noting it here though in case anyone else spots it.

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @CzarekDryl thanks for helping me fix my issue.

I got it to work and have some feedback.

1. When I'm only paying 1 user, the metadata is showing 'multiple recipients and tokens' which isn't accurate. It should show the number here when only 1 member is being paid.
image

2. The table looses it's styling in the completed state.
image

There should be NO horizontal rules between members at any stage (during the action creation and completed states) but only between the last row and the total field as per Figma below: 

image


3. The mobile total field at the bottom of the table has padding issues:
image
 
 Figma showing correct padding:
 
image

4. Unverified members are missing their alert icon showing in the table.
image


5. Size of the verified icon and the padding after the username should match Figma on desktop and mobile - it should match the same size as the building state before the CTA is selected.
image

Link to example: https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-v5?type=design&node-id=3643-417416&mode=design&t=GU7PGCjsxu5u5B0A-4

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Thanks for your work so far! I've noticed a couple of issues:

It isn't possible to change the token type on mobile.

https://github.com/JoinColony/colonyCDapp/assets/34915414/3556a530-8c49-488a-a825-fa809de6bc96

On the completed view, the horizontal rules should continue the entire way across. The total number of tokens should also be right aligned.

Screenshot 2024-03-19 at 10 36 24

Figma:

Screenshot 2024-03-19 at 10 36 36

**Comment by CzarekDryl** [comment]
@iamsamgibbs First one is out of scope because in this issue I'm taking care of completed action.

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Codewise it looks good 👍 

I'm still able to cause some overlay issues however, and there is a weird horizontal scrollbar in the user popover (which doesn't happen elsewhere this component is used):

Screenshot 2024-03-19 at 18 57 10

image

**Comment by CzarekDryl** [comment]
@jakubcolony Currently, I have no idea how to improve the table, it would be best if Melynda came up with something. Can we merge it into a feature branch and then fix it, because we are seriously blocked by this task.

**Comment by arrenv** [comment]
Adding a comment here regarding the table at table view, the solution to this can just be to toggle the mobile version of the table at this size.

I think this is worth it's own issue though, so, I have created an issue here https://github.com/JoinColony/colonyCDapp/issues/2072 and this PR, so that this PR can be completed without including this fix.

**Review by melyndav — CHANGES_REQUESTED** [review]
@CzarekDryl all looks good to go, I've approved it. :)

**Review by jakubcolony — APPROVED** [review]
LGTM code-wise, good to merge once Mel's comments are addressed 👍

**Review by iamsamgibbs — APPROVED** [review]

**Pull request merged**
